Fails to build with error: ‘memcpy’ was not declared in this scope

Building with gcc 6, `make` fails with:

```
guetzli.cc
guetzli/guetzli.cc: In function ‘bool {anonymous}::ReadPNG(FILE*, int*, int*, std::vector<unsigned char>*)’:
guetzli/guetzli.cc:119:45: error: ‘memcpy’ was not declared in this scope
         memcpy(row_out, row_in, 3 * (*xsize));
                                             ^
guetzli/guetzli.cc: In function ‘int main(int, char**)’:
guetzli/guetzli.cc:221:68: error: ‘memcmp’ was not declared in this scope
       memcmp(in_data.data(), kPNGMagicBytes, sizeof(kPNGMagicBytes)) == 0) {
                                                                    ^
guetzli.make:165: recipe for target 'obj/Release/guetzli.o' failed
make[1]: *** [obj/Release/guetzli.o] Error 1
Makefile:28: recipe for target 'guetzli' failed
make: *** [guetzli] Error 2
```

--- a/guetzli.make
+++ b/guetzli.make
@@ -22,9 +22,9 @@ ifeq ($(config),release)
   ALL_CFLAGS += $(CFLAGS) $(ALL_CPPFLAGS) -O3
   ALL_CXXFLAGS += $(CXXFLAGS) $(ALL_CFLAGS) -std=c++11
   ALL_RESFLAGS += $(RESFLAGS) $(DEFINES) $(INCLUDES)
-  LIBS +=
+  LIBS += -lpng -lgflags_nothreads -lz -lpthread
   LDDEPS +=
-  ALL_LDFLAGS += $(LDFLAGS) -s -lpng -lgflags_nothreads -lz
+  ALL_LDFLAGS += $(LDFLAGS) -s
   LINKCMD = $(CXX) -o "$@" $(OBJECTS) $(RESOURCES) $(ALL_LDFLAGS) $(LIBS)
   define PREBUILDCMDS
   endef
@@ -49,9 +49,9 @@ ifeq ($(config),debug)
   ALL_CFLAGS += $(CFLAGS) $(ALL_CPPFLAGS) -g
   ALL_CXXFLAGS += $(CXXFLAGS) $(ALL_CFLAGS) -std=c++11
   ALL_RESFLAGS += $(RESFLAGS) $(DEFINES) $(INCLUDES)
-  LIBS +=
+  LIBS += -lpng -lgflags_nothreads -lz -lpthread
   LDDEPS +=
-  ALL_LDFLAGS += $(LDFLAGS) -lpng -lgflags_nothreads -lz
+  ALL_LDFLAGS += $(LDFLAGS)
   LINKCMD = $(CXX) -o "$@" $(OBJECTS) $(RESOURCES) $(ALL_LDFLAGS) $(LIBS)
   define PREBUILDCMDS
   endef

--- a/guetzli/guetzli.cc
+++ b/guetzli/guetzli.cc
@@ -18,6 +18,7 @@
 #include <cstdlib>
 #include <memory>
 #include <string>
+#include <string.h>
 #include "gflags/gflags.h"
 #include "png.h"
 #include "guetzli/processor.h"

--- a/guetzli/jpeg_data_writer.cc
+++ b/guetzli/jpeg_data_writer.cc
@@ -18,6 +18,7 @@
 
 #include <assert.h>
 #include <cstdlib>
+#include <string.h>
 
 #include "guetzli/entropy_encode.h"
 #include "guetzli/fast_log.h"

--- a/guetzli/processor.cc
+++ b/guetzli/processor.cc
@@ -18,6 +18,7 @@
 
 #include <algorithm>
 #include <set>
+#include <string.h>
 #include <vector>
 
 #include "guetzli/butteraugli_comparator.h"

--- a/premake5.lua
+++ b/premake5.lua
@@ -13,7 +13,9 @@ workspace "guetzli"
     language "C++"
     includedirs { ".", "third_party/butteraugli" }
     filter "action:gmake"
-      linkoptions { "-lpng", "-lgflags_nothreads", "-lz" }
+      -- We add pthread, because gflags_nothreads depends on it (sic!) for some
+      -- versions of gflags.
+      links { "png", "gflags_nothreads", "z", "pthread" }
     filter "action:vs*"
       links { "shlwapi" }
     filter {}

